### PR TITLE
Gdsfactory8

### PR DESCRIPTION
**Pull request by joamatab**
- fix plugins for gdsfactory8

@yaugenst-flex
@nikosavola 
@simbilod

### CODE REVIEW

File: gplugins/gmsh/xy_xsection_mesh.py [hunk 1]
```diff
@@ -106,9 +105,8 @@ def xy_xsection_mesh(
         layer_port_delimiter: Delimiter to use for new layers generated for ports: "layer{delimiter}port_name".
     """
     if port_names:
-        mesh_component = gf.Component()
-        _ = mesh_component << union(component, by_layer=True)
-        mesh_component.add_ports(component.get_ports_list())
+        mesh_component = component
+        mesh_component.add_ports(component.ports)
```
Comment (simbilod) [inline gplugins/gmsh/xy_xsection_mesh.py:109]: I think mesh_component should be a new component, to which we add the polygons and ports of component

right now we get the following error on the test on these lines:

`ValueError: add_port() Port name 'o1' exists in 'Unnamed_57c8ad0b'`

**Review by simbilod — CHANGES_REQUESTED** [review]

File: gplugins/gmsh/xy_xsection_mesh.py (hunk 1 above)
Comment (joamatab) [reply gplugins/gmsh/xy_xsection_mesh.py:109]: i see, yes, we can make a new component

Comment (joamatab) [reply gplugins/gmsh/xy_xsection_mesh.py]: does the new changes work?

Comment (simbilod) [reply gplugins/gmsh/xy_xsection_mesh.py]: is it running using gdsfactory8? We're getting `'Component' object has no attribute 'dbbox'` everywhere

Comment (joamatab) [reply gplugins/gmsh/xy_xsection_mesh.py]: i think i fixed this one

**Comment by yaugenst-flex** [comment]
Note that the fix from #421 _is not complete_! It just always puts ports on the `core` layer. This still needs fixing!

**Comment by joamatab** [comment]
Sounds good Yannick, the tests and docs are still failing

**Comment by simbilod** [comment]
> Sounds good Yannick, the tests and docs are still failing

The plugin will probably require a similar fix to the meshing, i.e. with the new layerstack now the polygons are returned from the AbstractLayer via a `get_shapes` method that returns a KLayout region.

See 347683ea1524493ed8c46f38f20966f745363293

**Comment by yaugenst** [comment]
Oops. probably should be using this one :sweat_smile: 

Anyways, yeah thanks @simbilod for the insight, I'll have another look tomorrow. It seemed ok on the simple components that I was running with.. let's see. I do rely on you for the meshing though so I assumed that part already worked ;)

**Pull request merged**